### ISSUE TITLE
suport gnome-shell 3.16

### ISSUE THREAD

**Issue by dffischer** [mathematicalcoffee/Gnome-Shell-Window-Buttons-Extension#23]
Gnome-shell 3.16 is out. I tried to add the version number to the array in [metadata.json](https://github.com/mathematicalcoffee/Gnome-Shell-Window-Buttons-Extension/blob/master/window_buttons%40biox.github.com/metadata.json), but this is not sufficient to make it run. Loading the extension fails with the message

    Extension "window_buttons@biox.github.com" had error: Error: Expected type interface for Argument 'file' but got type 'string'

Unfortunately, it does not tell on which code line the error occurred.

**Comment by l300lvl** [comment]
referencing this. i shall peek

**Comment by l300lvl** [comment]
@dffischer if you reference #22 you will see this for 3.14 https://github.com/danielkza/Gnome-Shell-Window-Buttons-Extension

it should work fine on 3.16 as well...

**Comment by dffischer** [comment]
Pulling in [danielkza's fork](https://github.com/danielkza/Gnome-Shell-Window-Buttons-Extension) indeed helps fixing #24. Unfortunately, the error mentioned here still stays present.

**Comment by danielkza** [comment]
Fedora 22 is out now and I'm on 3.16, I'll see if I can get things working.

**Comment by danielkza** [comment]
@dffischer  I think I got it working, just had to integrate your fix and change the stylesheet loading/unloading to use file objects instead of string paths. Check if it's fine for you in my fork: danielkza/Gnome-Shell-Window-Buttons-Extension

**Comment by l300lvl** [comment]
Confirmed here, working great on 3.16 now. 

Also, the setting "When should the buttons appear", options  "windows visible" and "any window focused" work great when combined with Show Desktop Button. This is a great addition if you use my extension as well, but it's awesome this is working again fully. Sorry I couldn't be of more use, I actually forgot about this. :+1:

**Comment by l300lvl** [comment]
@danielkza don't forget to update the metadata for 3.16 inclusion.

**Comment by danielkza** [comment]
Yeah, I just did that and a couple of other changes. I'll uploaded it to extensions.gnome.org as well. I'm fine with maintaining it from now on, since everyone else seems to be too busy nowadays.

**Comment by l300lvl** [comment]
@danielkza You don't have issue handling setup on your copy of this and it would be more spamish of me to have posted a new bug, but check your email please. I sent you some info that may be helpful.
